autogen/main.tf

```diff
@@ -27,7 +27,7 @@ data "google_compute_zones" "available" {
   {% endif %}
 
   project = var.project_id
-  region  = var.region
+  region  = var.region ? var.region : join("-", slice(split("-", var.zones[0]), 0, 2))
```

Comment:
Let's capture this logic as a `local` instead of duplicating between `main.tf` and `networks.tf`.

Reply:
Logic moved to `local` section.